leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/impl/CaliforniumLwM2mClientRequestSender.java

```diff
@@ -224,13 +218,21 @@ public void onReject() {
 
         public T waitForResponse() {
             try {
-                final boolean latchTimeout = latch.await(timeout, TimeUnit.MILLISECONDS);
-                if (!latchTimeout || coapTimeout.get()) {
+                boolean timeEllapsed = false;
+                if (timeout != null) {
+                    timeEllapsed = !latch.await(timeout, TimeUnit.MILLISECONDS);
+                } else {
+                    latch.await();
```

Comment:
Should we find a way to warn the user that this will now wait indefinitely?  Should we possibly put a maximum timeout of some very large number (say 60 minutes?) or some way for the user to unblock the client?  Also if the stop() method is called is there a way to have all of these latches possibly released?